core/src/main/java/org/frankframework/filesystem/SftpFileSystem.java

```diff
@@ -191,10 +191,19 @@ public void createFolder(String folder) throws FileSystemException {
 		if(folderExists(folder)) {
 			throw new FileSystemException("Create directory for [" + folder + "] has failed. Directory already exists.");
 		}
+
 		try {
-			ftpClient.mkdir(folder);
-		} catch (SftpException e) {
-			throw new FileSystemException(e);
+			String[] folders = folder.split("/");
+			for(int i = 1; i < folders.length; i++) {
+				folders[i] = folders[i - 1] + "/" + folders[i];
+			}
+			for(String f : folders) {
+				if(f.length() != 0 && !folderExists(f)) {
+					ftpClient.mkdir(f);
+				}
+			}
+		} catch (SftpException | ArrayIndexOutOfBoundsException e) {
```

Comment:
Waar zou een ArrayIndexOutOfBoundsException kunnen onstaan?

Reply:
Tja, al het mogelijke afvangen? Nu dat het werkt is het niet meer nodig nee